tracer/src/Datadog.Trace/Propagators/W3CTraceContextPropagator.cs

```diff
@@ -150,11 +154,9 @@ internal static string CreateTraceStateHeader(SpanContext context)
                 sb.Append("dd=");
 
                 // sampling priority ("s:<value>")
-                var samplingPriority = SamplingPriorityToString(context.TraceContext?.SamplingPriority);
-
-                if (samplingPriority != null)
+                if (context.TraceContext?.SamplingPriority is { } samplingPriority)
                 {
-                    sb.Append("s:").Append(samplingPriority).Append(TraceStateDatadogPairsSeparator);
+                    sb.Append("s:").Append(SamplingPriorityValues.ToString(samplingPriority)).Append(TraceStateDatadogPairsSeparator);
```

Comment:
> it's doesn't allocate the intermediate string

This is not calling `Int32.ToString()`, if that's what you're thinking. It returns string constants (for known values). Maybe I should rename that static `SamplingPriorityValues.ToString()` to avoid confusion.

https://github.com/DataDog/dd-trace-dotnet/blob/38903e8975c4927913c8ed9d394c782897a7f1f4/tracer/src/Datadog.Trace/SamplingPriorityValues.cs#L49-L60

Reply:
> This is not calling Int32.ToString()

Sure, except, it kinda is, you just hard coded some of the values 😅

Reply:
i.e. I can't see a value of `int` for which `SamplingPriorityValues.ToString()` gives something _different_ 😉

Reply:
But for all the known values, we use a constant. It's not allocating a new string, and it also avoids any other int-to-string conversion code when using `StringBuilder.Append(int)` (like globalization).

Reply:
This whole conversation is significantly longer than I expected 😅 I wasn't really suggesting you change it 😄 

But seeing as we're here already...

> It's not allocating a new string...
> In other words, we're comparing about StringBuilder.Append(int) vs StringBuilder.Append(string) where both arguments are constants.

Correct, for _known_ values. For not-known values I wins.

Ok, pedantry over. Lets move on, we can both agree nothing needs to change here 😅